src/instrumentsscene/view/instrumentspaneltreemodel.cpp

```diff
@@ -48,13 +48,10 @@ static QString notationToKey(const INotationPtr notation)
 }
 
 InstrumentsPanelTreeModel::InstrumentsPanelTreeModel(QObject* parent)
-    : QAbstractItemModel(parent)
+    : QAbstractItemModel(parent), m_selectionModel(new ItemMultiSelectionModel(this))
 {
     m_partsNotifyReceiver = std::make_shared<async::Asyncable>();
 
-    m_selectionModel = new ItemMultiSelectionModel(this);
-    m_selectionModel->setAllowedModifiers(Qt::ShiftModifier);
```

Comment:
Does moving instruments up and down still work correctly if there is a non-contiguous selection?

Reply:
It didn't at that time. It got all messed up.

I don't know about the present condition. But I think I will remove that commit. I don't think the problem of not being able to select instrument is too big (@Tantacrul hope you agree)

Reply:
I believe I had long ago created an algorithm for moving non-contiguous selections up and down. It was aimed at the parts dialog (never got merged into there though), but could probably be adapted for the instruments panel. I'll try to reuse it after this PR gets merged. But for now, let's indeed drop the commit from this PR.